commit 2fadaef41283aad7100fa73f01998cddaca25833
Author: Al Viro <viro@zeniv.linux.org.uk>
Date:   Tue Feb 15 22:52:11 2011 -0500

    auxv: require the target to be tracable (or yourself)
    
    same as for environ, except that we didn't do any checks to
    prevent access after suid execve
    
    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
    [dannf: backported to Debian's 2.6.26]

commit bf8db462fc178f51a71fcf01a0cbe9d51215f0bf
Author: dann frazier <dannf@debian.org>
Date:   Wed Aug 31 22:11:15 2011 -0600

    bugfix/all/auxv-require-the-target-or-self-to-be-traceable.patch

diff --git a/fs/proc/base.c b/fs/proc/base.c
index f582fff..fc5f0d7 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -300,9 +300,9 @@ out:
 
 static int proc_pid_auxv(struct task_struct *task, char *buffer)
 {
-	int res = 0;
-	struct mm_struct *mm = get_task_mm(task);
-	if (mm) {
+	struct mm_struct *mm = mm_for_maps(task);
+	int res = PTR_ERR(mm);
+	if (mm && !IS_ERR(mm)) {
 		unsigned int nwords = 0;
 		do
 			nwords += 2;
